### PR TITLE
Securing microservices initial draft

### PR DESCRIPTION
**Pull request by ManasiGandhi**

### CODE REVIEW

File: ref/general/securing-microservices.adoc [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+As applications move from the monolithic to microservices based environment, the security scenario needs to change too.
+Network security provides security at an external level as a first line of defense.
+However, in a microservices environment your applications need to be secured internally at a granular level.
+
+== Using an API gateway
+
+Microservices communicate with each other through APIs.
+Every point of communication offers vulnerabilities to security.
+Security needs to be implemented at the access level through authentication and authorization
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: simplify- also add period

`Security is implemented at the access level through authentication and authorization.`

you might need to add another sentence or two to specify what you mean by authentication (determine who a user is) and authorization (determine what roles user is in)

File: ref/general/securing-microservices.adoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+An API Gateway provides an entry point for client requests and protects the microservices from the client.
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: Authn and Authz are not specific to an API gateway. If you start this section w/ those, need to explain how an API gateway manages.

also might be better to say protects microservices from security threats. In general, the client should not be the threat (even though sometimes they are...)

File: ref/general/securing-microservices.adoc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Thus the client does not have any access to the microservices.
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: I understand what you're getting at but this makes it sound like no one can interact with your application. Expand on this idea in a couple sentences to make clear that the gateway manages access- not keeps everyone out

File: ref/general/securing-microservices.adoc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+== Monitoring security and tracing
+
+Monitoring microservices is complex, and end to end monitoring  is important.
+Automation in testing the containers is helpful in a microservices context.
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: explain briefly what you mean by testing containers. Try not to use "Automation" as the subject of the sentence.

File: ref/general/securing-microservices.adoc [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+Prometheus is an open source monitoring platform for microservices and a part of the Cloud Native Computing Foundation
+Distributed Tracing offers a solution in tracking an troubleshooting points of failure in a system
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: add a sentence or two about tracing- how does it track and troubleshoot points of failure?

avoid wrapping the verb in a clause- "offers a solution in tracking and troublshooting..." 
something like "is a system to track and troubleshoot..."

File: ref/general/securing-microservices.adoc [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+:seo-description: Securing cloud-native applications is an important part of your development process as applications move from monolithic to microservices architecture.
+:page-layout: general-reference
+:page-type: general
+= Securing cloud-native microservices
+
+Securing cloud-native applications is an important part of your development process as applications move from monolithic to microservices architecture.
+Rather than being an afterthought, security needs to take precedence in the development cycle.
+There are several ways in which you can ensure the security of your cloud-native microservices.
+
+== Defense in-depth
```
[the remark this answers is not in the document]
Comment (ManasiGandhi) [reply ref/general/securing-microservices.adoc]: IBM uses it without any hyphens, so taking away the hyphens.

File: ref/general/securing-microservices.adoc [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+A simple firewall does not provide adequate security in a microservices context.
+You need to apply multiple layers of security to ensure that your application is safe.
+In a complex  and distributed microservices environment, where multiple processes exist, a single point security at the network level is inadequate.
+Processes like authentication and authorization, provide defense in depth security.
```
[the remark this answers is not in the document]
Comment (ManasiGandhi) [reply ref/general/securing-microservices.adoc]: IBM uses it without any hyphens, when used as a noun.

File: ref/general/securing-microservices.adoc [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+== Addressing the changed scenario
+
+As applications move from the monolithic to microservices based environment, the security scenario needs to change too.
+Network security provides security at an external level as a first line of defense.
+However, in a microservices environment your applications need to be secured internally at a granular level.
+
+== Using an API gateway
+
+Microservices communicate with each other through APIs.
+Every point of communication offers vulnerabilities to security.
```
[the remark this answers is not in the document]
Comment (ManasiGandhi) [reply ref/general/securing-microservices.adoc]: I added 'entails'. Hope doesn't sound too complicated.

File: ref/general/securing-microservices.adoc [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+As applications move from monolithic to microservices architecture, securing cloud-native applications is an important part of your development process.
+Rather than being an afterthought, security needs to take precedence in the development cycle.
+There are several ways to ensure the security of your cloud-native microservices.
+
+== Defense in depth
+
+A simple firewall does not provide adequate security in a microservices context.
+You need to apply multiple layers of security to ensure that your application is safe.
+In a complex and distributed microservices environment, where multiple processes exist, a single point of security at the network level is inadequate.
+Processes like authentication and authorization, hashing passwords, and encryption provide defense in depth security.
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: defense-in depth security

In this case, "defense-in-depth" is a compound adjective modifying "security"

File: ref/general/securing-microservices.adoc [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+
+== The changed scenario
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: this heading could tell you more of what the section is about. As is, it seems indistinct

File: ref/general/securing-microservices.adoc [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+As applications move from the monolithic to microservices based architecture, the security scenario evolves accordingly.
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: hyphenate compunad adjective: "microservices-based"

File: ref/general/securing-microservices.adoc [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+Network security provides a first line of defense at an external level.
+However, in a microservices environment, your applications also need internal security between services.
+
+== Application Programming Interface (API) gateway for security
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: should this be an H3 heading? Is it a subsection of The changed scenario?

File: ref/general/securing-microservices.adoc [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+Microservices communicate with each other through APIs.
+Every point of communication entails vulnerabilities to security.
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: change "vulnerabilities to security" to "security vulnerabilities"

File: ref/general/securing-microservices.adoc [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+An API Gateway provides an entry point for client requests and protects the microservices from the client through authentication and authorization.
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: ok to add later but need to explain what an API gateway is

File: ref/general/securing-microservices.adoc [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+Thus, the client does not have any access to the microservices.
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: the client does have access to the microservices- only through the gateway. Might be more correct to say the client does not interact with the microservices directly

File: ref/general/securing-microservices.adoc [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+
+== Monitoring security and tracing
+
+Monitoring microservices is complex, and end to end monitoring  is important.
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: Monitoring microservices is complex, and end-to-end monitoring  is important.

File: ref/general/securing-microservices.adoc [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+Prometheus is an open source monitoring platform for microservices and a part of the Cloud Native Computing Foundation
```
Comment (dmuelle) [inline ref/general/securing-microservices.adoc]: add period after "Foundation"

File: ref/general/securing-microservices.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (dmuelle) [reply ref/general/securing-microservices.adoc]: In this case, "defense-in-depth" is a compound adjective modifying "security"

File: ref/general/securing-microservices.adoc (hunk 14 above)
Comment (ManasiGandhi) [reply ref/general/securing-microservices.adoc]: Sure

File: ref/general/securing-microservices.adoc (hunk 12 above)
Comment (ManasiGandhi) [reply ref/general/securing-microservices.adoc]: Still working on it. Right now, building on the meeting discussion bullet points.

**Pull request merged**